dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/repository/impl/SessionDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.dao.repository.impl;
+
+import org.apache.dolphinscheduler.dao.entity.Session;
+import org.apache.dolphinscheduler.dao.mapper.SessionMapper;
+import org.apache.dolphinscheduler.dao.repository.BaseDao;
+import org.apache.dolphinscheduler.dao.repository.SessionDao;
+
+import java.util.List;
+
+import lombok.NonNull;
+
+import org.springframework.stereotype.Repository;
+
+import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
+
+@Repository
+public class SessionDaoImpl extends BaseDao<Session, SessionMapper> implements SessionDao {
+
+    public SessionDaoImpl(@NonNull SessionMapper sessionMapper) {
+        super(sessionMapper);
+    }
+
+    public void deleteByUserId(Integer userId) {
+        mybatisMapper.delete(new QueryWrapper<>(Session.builder().userId(userId).build()));
+    }
+
+    @Override
+    public List<Session> queryByUserId(Integer userId) {
+        return mybatisMapper.selectList(new QueryWrapper<>(Session.builder().userId(userId).build()));
```

Comment:
Can we avoid to use wrapper? Although this is a simple query looks fine, many complex query wrapper may be written later. 
Use mapper.xml uniformly is a better way, WDYT?

Reply:
Agree with you, we shouldn't expose wrapper to service. I change this method to 
```
/**
   * Query the entity by condition.
   */
  List<Entity> queryByCondition(Entity queryCondition);
```
And add implementation in BaseDao
```
@Override
    public List<ENTITY> queryByCondition(ENTITY queryCondition) {
        if (queryCondition == null) {
            throw new IllegalArgumentException("queryCondition can not be null");
        }
        return mybatisMapper.selectList(new QueryWrapper<>(queryCondition));
    }
```

Then we can avoid using wrapper in` dao`, then we can use this method for simple query, and remove a lot of method like `queryByCode`, `queryByName`, `queryByXX`, for this kind of query we can directly use dao.queryByCondition(entity).

Reply:
It seems good.